### PR TITLE
Update to Agroal 2.3

### PR DESCRIPTION
**Pull request by barreiro**

### CODE REVIEW

File: extensions/hibernate-orm/deployment/src/test/resources/application-multiple-persistence-units-annotations.properties [hunk 1]
```diff
@@ -3,8 +3,10 @@ quarkus.datasource.jdbc.max-size=1
 
 quarkus.datasource.users.db-kind=h2
 quarkus.datasource.users.jdbc.min-size=2
+quarkus.datasource.users.jdbc.transactions=xa
```
Comment (yrodiere) [inline extensions/hibernate-orm/deployment/src/test/resources/application-multiple-persistence-units-annotations.properties:6]: I don't think this test is supposed to involve XA transactions... And regardless, we probably don't want users to have to change their configuration unless there's a good reason.

Why this change?

Same for the other tests.

**Comment by geoand** [comment]
@barreiro thanks!

Can you describe why `quarkus.datasource.users.jdbc.transactions=xa` is needed?

**Review by yrodiere — COMMENTED** [review]
Hey. A few comments below.

In case you're unaware, you won't get CI test runs unless you mark your PR as ready for review. But maybe you're testing it on your fork?

**Comment by barreiro** [comment]
@geoand we added `LastResource` to `LocalXAResource` (https://github.com/agroal/agroal/blob/2.3/agroal-narayana/src/main/java/io/agroal/narayana/LocalXAResource.java#L23). The consequence being that at most one datasource can be enlisted, and if you think about it it's not safe two have two or more.

I will agree that the error is not descriptive of what's going on.

@yrodiere thanks. will move out of draft to run CI.

**Comment by gastaldi** [comment]
@barreiro so the rule is that when you have more than one DataSource declared they should always be XA? Seems like a breaking change to me

That may need some documentation changes and perhaps a better error message, as the `Unable to enlist connection to existing transaction` is not really intuitive IMHO

**Review by yrodiere — CHANGES_REQUESTED** [review]
I can understand having to use XA when multiple datasources are involved in the same transaction, but if the datasources are always used in separate transactions, it feels like a newly introduced limitation more than anything else...

Is this configuration change required even for people who never use more than one datasource in the same transaction? If so, that's 1. hard to justify and 2. a _huge_ breaking change. Basically every application using multiple datasources will be affected, and will need to start using XA, and I'm sure this will put off many people.

We need to discuss this in more details. And ideally, find a solution that doesn't require everyone to change their configuration.

**Comment by gsmet** [comment]
Yeah, I also am against enabling XA for everything. Most of our users using several datasources won't involve several datasources in the same transaction.

**Comment by barreiro** [comment]
@gastaldi @yrodiere @gsmet the exception only occurs when enlisting the second datasource. the failing tests do exactly that, mix both datasources on the same transaction. another option would be to refactor the tests to use two transactions, one for each datasource.

there was another quarkus issue, that I can't find at the moment, where this behavior was discussed and the conclusion was that we should'd allow two non-xa datasources on the same transaction, for obvious reasons.

**Comment by yrodiere** [comment]
> there was another quarkus issue, that I can't find at the moment, where this behavior was discussed and the conclusion was that we should'd allow two non-xa datasources on the same transaction, for obvious reasons.

No argument there.

> the exception only occurs when enlisting the second datasource

Great, we're on the same page then.

> another option would be to refactor the tests to use two transactions, one for each datasource.

I think that's the way to go. Unless I'm mistaken, none of these tests were intended to test usage of two datasources in the same transaction. If they do, that's most likely just because that was convenient.

**Comment by gastaldi** [comment]
Kind reminder to please document this change in the Migration Guide and in the guides.

**Comment by barreiro** [comment]
@yrodiere fixed the test cases

**Review by yrodiere — APPROVED** [review]
LGTM, thanks.

We might want to use `QuarkusTransaction` instead, to get proper rollbacks on failures, but... I guess that'll be for whoever debugs tests next to figure out :)

**Pull request merged**